Makefile

```diff
@@ -163,7 +164,10 @@ dist: writeversion staticdeps staticdeps-cext buildconfig assets translation-dja
 pex: writeversion
 	ls dist/*.whl | while read whlfile; do pex $$whlfile --disable-cache -o dist/kolibri-`cat kolibri/VERSION | sed 's/+/_/g'`.pex -m kolibri --python-shebang=/usr/bin/python; done
 
-translation-extract: assets
+translation-extract: clean-build
+	@echo ""
+	@echo "!! This assumes that you are running with latest assets. Run 'make assets' if in doubt !!"
+	@echo ""
```

Comment:
This change has left our documentation out-of-date, and the command itself is no longer named correctly.

ref: #4228

Reply:
Sorry that you had issues debugging this and that changes left inaccurate information in the documentation. I misunderstood what was going on, thought that `makemessages` was still doing extractions for frontend at this point, but I can see that everything seems to be handled in `frontend_build/src/extract_$trs.js`.

Maybe we should move this out of assets building and into `translations-extract` since it's only essential at this point? I mean by adding `node frontend_build/src/extract_\$trs.js` in the target?

Reply:
It is useful for it to happen dynamically during asset building as that keeps the frontend messages meaningful, plus the webpack process allows us to automatically identify which strings should be extracted and which plugin they are associated with.

In short, no.

Reply:
Oh, it is a pre-requisite for messages to display in English (the source language) ?

Reply:
Yes, otherwise the message id is used!

Reply:
Ah, okay, that's no good - and we need to use these Message IDs instead of the English source strings?

Reply:
Yeah, the source strings can get quite complex and having them referenced by id means we don't have to go looking for them in the source of the templates!

Reply:
Ah, okay, so putting all this together: There is no way to extract frontend messages separately from the assets building, this has to be done in `make assets`?

Reply:
Yes, unless we want to do some fairly extensive code refactoring :)

Reply:
Nah, let's skip that :) I think it's a part of the architecture of message extraction in JS: It cannot be detected by an outside toolchain, but relies on the runtime environment (which kind of becomes part of the toolchain).